integration/spark-common/src/main/java/org/apache/carbondata/spark/util/SparkDataTypeConverterImpl.java

```diff
@@ -171,4 +176,21 @@ public Object wrapWithGenericRow(Object[] fields) {
     }
     return fields;
   }
+
+  public static StructType convertToSparkSchema(ColumnSchema[] carbonColumns) {
```

Comment:
I think this method do not handle complex datatype schema. Can you check `CarbonRelation` how we are handling complex schema and same way can be used here

Reply:
fixed